Lib/test/test_sqlite3/test_types.py

```diff
@@ -103,27 +103,29 @@ def test_string_with_surrogates(self):
         row = self.cur.fetchone()
         self.assertIsNone(row)
 
-    @unittest.skipUnless(sys.maxsize > 2**32, 'requires 64bit platform')
-    @support.bigmemtest(size=2**31, memuse=4, dry_run=False)
+    @unittest.skipUnless(sys.maxsize > 2**64, 'requires (s)size_t > 64bit')
+    @support.bigmemtest(size=2**64+1, memuse=4, dry_run=False)
```

Comment:
No, it is not the purpose of these tests. It does not test that the code raises an expected exception. It tests that the code does not corrupt the data, does not raise unexpected exception and does not crash. What would happen if remove the overflow check.

How do existing tests fail with these changes applied? Do they stop raising DataError? It is good, we should change tests for this.

Reply:
> It does not test that the code raises an expected exception.

The test executes `assertRaises`; it *does test that an exception is raised*. Yes, that also **implies that no corruption has happened and that sqlite3 did not segfault**.

The current test code does this:

1. assert that DataError is raised when binding very large strings or buffers
2. select rows from the test table and assert that no row was inserted

IOW, it verifies that sqlite3 raised an exception for large input, and that no rows were added.

Reply:
Well, now the behavior is changed, it was not unintentional, and the tests should be changed as well.

Reply:
> [...] the tests should be changed as well.

That is part of the aim of this PR.

Reply:
What happens if left these tests as is?

Reply:
Well, on _my_ machine, they actually succeed (though I have to remove the bigmem decorators in order to force them to run), so I guess they are already broken.